Wikipedia/Categories/MWKSection+DisplayHtml.m

```diff
@@ -22,13 +22,20 @@ - (NSString*)displayHTML:(NSString*)html {
 }
 
 - (NSString*)getHeaderTag {
+    NSUInteger headingSize = [self getHeadingTagSize];
     return
-        [NSString stringWithFormat:@"<h1 class='section_heading' id='%@' sectionId='%d'>%@</h1>%@",
+        [NSString stringWithFormat:@"<h%ld class='section_heading' id='%@' sectionId='%d'>%@</h%ld>%@",
+         headingSize,
          self.anchor,
          self.sectionId,
          [self isLeadSection] ? self.title.text : self.line,
+         headingSize,
          [self isLeadSection] ? MWKSectionDisambigAndPageIssuesPlaceholderDiv : @""
         ];
 }
 
+- (NSInteger)getHeadingTagSize {
+    return MIN(MAX(self.level.integerValue, 1), 6);
+}
```

Comment:
Instead of just using *H1* tags for all section headers, use the full range of *H1* through *H6* according to level depth.